doc/apdu.md

```diff
@@ -0,0 +1,432 @@
+
+| *CLA*  | *INS*  | *P1*   | *P2* |
+|--------|--------|--------|------|
+| `0x80` | `0x01` | `0x00` | `P2` |
+
+Requests authorization to bake with the key associated with the given
+`path` and `P2`.
+
+If no `path` is provided, the request is performed using the key
+already authorized.
+
+If the request is accepted, the key is defined as the [`authorized-key`](ram.md#authorized-key)
+and the public key is returned.
+
+Refuse the request do not erase the existing [`authorized-key`](ram.md#authorized-key)
+
+#### Input data
+
+| Length       | Description               |
+|--------------|---------------------------|
+| `<variable>` | The `path` (can be empty) |
+
+#### Output data
+
+| Length     | Description             |
+|------------|-------------------------|
+| `1`        | The public key `length` |
+| `<length>` | The public key          |
+
+### `GET_PUBLIC_KEY`
+
+| *CLA*  | *INS*  | *P1*   | *P2* |
+|--------|--------|--------|------|
+| `0x80` | `0x02` | `0x00` | `P2` |
+
+Get the public key according to the `path` and `P2`.
+
+Not allowed for permissionless legacy comm in browser.
+
+#### Input data
+
+| Length       | Description |
+|--------------|-------------|
+| `<variable>` | The `path`  |
+
+#### Output data
+
+| Length     | Description             |
+|------------|-------------------------|
+| `1`        | The public key `length` |
+| `<length>` | The public key          |
+
+### `PROMPT_PUBLIC_KEY`
+
+| *CLA*  | *INS*  | *P1*   | *P2* |
+|--------|--------|--------|------|
+| `0x80` | `0x02` | `0x00` | `P2` |
+
+Requests to get the public key according to the `path` and `P2`.
+
+#### Input data
+
+| Length       | Description |
+|--------------|-------------|
+| `<variable>` | The `path`  |
+
+#### Output data
+
+| Length     | Description             |
+|------------|-------------------------|
+| `1`        | The public key `length` |
+| `<length>` | The public key          |
+
+### `SIGN`
+
+#### First apdu
+
+| *CLA*  | *INS*  | *P1*             | *P2* |
+|--------|--------|------------------|------|
+| `0x80` | `0x04` | `0x00` or `0x80` | `P2` |
+
+Set the signing key to the key associated with the given `path` and
+`P2`.
+
+This step is not required, as long as the [`authorized-key`](ram.md#authorized-key) has been
+defined. In this case the signature will be performed by this
+[`authorized-key`](ram.md#authorized-key).
+
+##### Input data
+
+| Length       | Description |
+|--------------|-------------|
+| `<variable>` | The `path`  |
+
+##### Output data
+
+No output data.
+
+#### Other apdus
+
+| *CLA*  | *INS*  | *P1*             | *P2* |
+|--------|--------|------------------|------|
+| `0x80` | `0x04` | `0x01` or `0x81` | `__` |
+
+Request to sign the `message`.
+
+Use `P1 = 0x81` to indicate that the message has been fully sent.
+
+Once the message has been fully sent and the request has been
+accepted, the signature of the message is returned.
+
+Messages sent in more than one packet will be refused.
```

Comment:
Then why there is a 0x01 and 0x81, if more than one packet can not be used for messages.

Reply:
Issue: https://github.com/trilitech/ledger-app-tezos-baking/issues/61